### PR TITLE
fix(integrations): StacktraceLinkModal deprecation warnings

### PR DESCRIPTION
**Pull request by mgaeta**
Fixes [API-1642](https://getsentry.atlassian.net/browse/API-1642). We're seeing a lot of deprecation warning in tests caused by `AsyncComponent`. This PR moves async logic out of the Stacktrace Linking modal.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx [hunk 1]
```diff
@@ -173,7 +213,7 @@ class StacktraceLink extends AsyncComponent<Props, State> {
                 deps =>
                   this.project && (
                     <StacktraceLinkModal
-                      onSubmit={() => this.handleSubmit()}
+                      onSubmit={() => this.handleSubmit}
```
Comment (scefali) [inline src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx:216]: @mgaeta hmmm this seems strange, are we actually invoking `handleSubmit` here? I'm also confused why changing stack trace linking is part of a PR about  the warnings caused by`getEndPointString`.

**Comment by scefali** [comment]
@mgaeta I'm confused about the purpose of the PR. I see that the method `getEndpoint` is a deprecated method on `AsyncComponent`. But `getEndpoint` is not used in either form. There's a `getEndPointString` method but that's different.

**Comment by mgaeta** [comment]
I just split this into two PRs in order to make it easier to review. See https://github.com/getsentry/sentry/pull/23639.

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx [hunk 2]
```diff
@@ -134,9 +135,48 @@ class StacktraceLink extends AsyncComponent<Props, State> {
     }
   }
 
-  handleSubmit() {
+  handleSubmit = async (sourceCodeInput, filename) => {
+    const {organization} = this.props;
+    const project = this.project as any;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx:140]: Should this be `Project` type?

**Review by billyvg — COMMENTED** [review]
I believe this changes the behavior slightly by moving the api requests up to the parent component - after submitting, the modal will always close regardless of the result of the POST request.

If you prefer the old behavior, you can use `withApi` to pass the api client to `stacktraceLinkModal`.

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**